docs/blockentities/index.md

```diff
@@ -1,141 +1,239 @@
 :::
 
-## Ticking `BlockEntities`
-
-If you need a ticking `BlockEntity`, for example to keep track of the progress during a smelting process, another method must be implemented and overridden within `EntityBlock`: `EntityBlock#getTicker(Level, BlockState, BlockEntityType)`. This can implement different tickers depending on which logical side the user is on, or just implement one general ticker. In either case, a `BlockEntityTicker` must be returned. Since this is a functional interface, it can just take in a method representing the ticker instead:
+Finally, we need to modify the block class associated with the block entity. This means that we will not be able to attach block entities to simple instances of `Block`, instead, we need a subclass:
 
 ```java
-// Inside some Block subclass
-@Nullable
-@Override
-public <T extends BlockEntity> BlockEntityTicker<T> getTicker(Level level, BlockState state, BlockEntityType<T> type) {
-  return type == MyBlockEntityTypes.MYBE.get() ? MyBlockEntity::tick : null;
-}
-
-// Inside MyBlockEntity
-public static void tick(Level level, BlockPos pos, BlockState state, MyBlockEntity blockEntity) {
-  // Do stuff
+// The important part is implementing the EntityBlock interface and overriding the #newBlockEntity method.
+public class MyEntityBlock extends Block implements EntityBlock {
+    // Constructor deferring to super.
+    public MyEntityBlock(BlockBehaviour.Properties properties) {
+        super(properties);
+    }
+
+    // Return a new instance of our block entity here.
+    @Override
+    public BlockEntity newBlockEntity(BlockPos pos, BlockState state) {
+        return new MyBlockEntity(pos, state);
+    }
 }
 ```
 
-:::note
-This method is called each tick; therefore, you should avoid having complicated calculations in here. If possible, you should make more complex calculations every X ticks. (The amount of ticks in a second may be lower then 20 (twenty) but won't be higher)
-:::
+And then, you of course need to use this class as the type in your block registration:
 
-## Synchronizing the Data to the Client
+```java
+public static final DeferredBlock<MyEntityBlock> MY_BLOCK_1 =
+        BLOCKS.register("my_block_1", () -> new MyEntityBlock( /* ... */ ));
+public static final DeferredBlock<MyEntityBlock> MY_BLOCK_2 =
+        BLOCKS.register("my_block_2", () -> new MyEntityBlock( /* ... */ ));
+```
 
-There are three ways of syncing data to the client: synchronizing on chunk load, on block updates, and with a custom network message.
+## Storing Data
 
-### Synchronizing on LevelChunk Load
+One of the main purposes of `BlockEntity`s is to store data. Data storage on block entities can happen in two ways: directly reading and writing [NBT][nbt], or using [data attachments][dataattachments]. This section will cover reading and writing NBT directly; for data attachments, please refer to the linked article.
```

Comment:
This should mention that data attachments are almost exclusively relevant to BEs from vanilla or other mods. For a mod's own BEs, direct storage in the BE itself should usually be preferred.